pkg/deploy/graph/analysis/dc.go

```diff
@@ -63,10 +63,12 @@ func ictMarker(g osgraph.Graph, f osgraph.Namer, dcNode *deploygraph.DeploymentC
 				}
 			}
 
-			if bcNode := buildedges.BuildConfigForTag(g, istNode); bcNode != nil {
-				// Avoid warning for the dc image trigger in case there is a build in flight.
-				if latestBuild := buildedges.GetLatestBuild(g, bcNode); latestBuild != nil && !buildutil.IsBuildComplete(latestBuild.Build) {
-					return nil
+			if bcNodes := buildedges.BuildConfigForTag(g, istNode); bcNodes != nil && len(bcNodes) > 0 {
+				for _, bcNode := range bcNodes {
+					// Avoid warning for the dc image trigger in case there is a build in flight.
+					if latestBuild := buildedges.GetLatestBuild(g, bcNode); latestBuild != nil && !buildutil.IsBuildComplete(latestBuild.Build) {
```

Comment:
You're returning a slice now, so it can't be `nil`, so you only need the `len(bcNodes)>0` check.